src/assets/ts/data_backend/index.ts

```diff
@@ -179,16 +179,24 @@ export class ClientSocketBackend extends DataBackend {
     this.clientId = Date.now() + '-' + ('' + Math.random()).slice(2);
   }
 
-  public async init(host: string, password: string, docname = '') {
-    this.events.emit('saved');
-
+  private async connect(host: string, password: string, docname: string) {
     logger.info('Trying to connect', host);
+    let that = this;
     this.ws = new WebSocket(`${host}/socket`);
     this.ws.onerror = (err) => {
-      throw new Error(`Socket connection error: ${err}`);
+      // throw new Error(`Socket connection error: ${err}`);
+      logger.info(`Socket connection error: ${err}`);
+      logger.info('Trying to reconnect...');
+      setTimeout(async function() {
```

Comment:
does it work if you do

`async () => {` instead of `async function() {` and then use `this` instead of `that`?

Reply:
I changed the syntax. That's better.